src/implementation/diagonal.jl

```diff
@@ -73,8 +70,8 @@ function tensorcontract!(C::AbstractArray, pC::Index2Tuple,
         C2 = StridedView(C3.parent, totsize, newstrides, C3.offset, C3.op)
     end
 
-    op1 = isone(α) ? (*) : (x, y) -> α * x * y
-    op2 = iszero(β) ? zero : isone(β) ? nothing : y -> β * y
+    op1 = (x, y) -> scale(x * y, α)
```

Comment:
Should probably check if it is worth it to do something like:
```julia
op1 = (x, y) -> Base.Fix2(scale, α)(x * y)
```

Reply:
I am not sure if using `Fix` is useful if you still have to create an anonymous function. It is useful if you want to create a one-arg function out of a 2-arg function by fixing one of the arguments.

Maybe `op1 = Base.Fix2(scale, α) ∘ *` ? Might be a bit cryptic though.

Reply:
I mostly included it here as a way of avoiding the whole captured variable issues, but I am not sure if this actually achieves that. I am pretty sure it does so for the 1-arg function out of a 2-arg function, but I don't know what `∘` does.

Reply:
`∘` is function composition. It does seem to create a specific (parametric) type so I think it is fine to combine this with `Fix2` and still avoids anonymous functions and captured variables.